### PR TITLE
SBT 7: Entity Properties

### PR DESCRIPTION
**Pull request by quantori-cherednichenko**

### CODE REVIEW

**Comment by quantori-cherednichenko** [comment]
TODO:
- add logging
- add code snippets to *.ipynb files

File: src/signals_notebook/entities/entity.py [hunk 1]
```diff
@@ -88,6 +150,47 @@ def _get_list_params(cls) -> Dict[str, Any]:
...
+
+        response = api.call(
+            method='GET',
+            path=(self._get_endpoint(), self.eid, 'properties'),
+        )
+        result = PropertiesResponse(**response.json())
+        properties = [cast(ResponseData, item).body for item in result.data]
+
+        for item in properties:
+            sample_property = cast(Property, item)
```
Comment (quantori-pokidovea) [inline src/signals_notebook/entities/entity.py:168]: sample?

File: src/signals_notebook/entities/entity.py [hunk 2]
```diff
@@ -88,6 +150,47 @@ def _get_list_params(cls) -> Dict[str, Any]:
...
+            assert sample_property.id
+
+            self._properties.append(sample_property)
+            self._properties_by_id[sample_property.id] = sample_property
+        log.debug('Properties in Entity: %s were reloaded', self.eid)
+
+    def update_properties(self, force: bool = True) -> None:
```
Comment (quantori-pokidovea) [inline src/signals_notebook/entities/entity.py:175]: Почему это не в методе save, а отдельно?

Comment (quantori-cherednichenko) [reply src/signals_notebook/entities/entity.py:175]: I have 2 problems with this case:
1. In our situation I have no idea how detect changes in attributes and save only them (without _properties). Or Should we reload _properties and attributes at the same time? 
2. We collect to '_properties' all properties with attributes which were collected after Entity creation.

I dont know how to solve this problem. In the worst case we can rewrite changed attributes if we will use _reload_properties and save methods at the same time.

File: src/signals_notebook/entities/entity.py [hunk 3]
```diff
@@ -297,7 +272,7 @@ def save(self, force: bool = True) -> None:
         log.debug('Updating properties in Entity: %s...', self.eid)
         if self._properties:
             [request_body.append(item.representation_for_update) for item in self._properties if item.is_changed]
-
+        print(request_body)
```
Comment (quantori-pokidovea) [inline src/signals_notebook/entities/entity.py:275]: Seriously?

**Review by quantori-pokidovea — APPROVED** [review]

**Pull request merged**